modules/react/tooltip/lib/useTooltip.tsx

```diff
@@ -135,7 +135,7 @@ export function useTooltip<T extends Element = Element>({
 
   const targetProps = {
     // extra description of the target element for assistive technology
-    'aria-describedby': type === 'describe' && visible ? id : undefined,
+    'aria-description': type === 'describe' ? titleText : undefined,
```

Comment:
I think this is potentially breaking change, because now tooltip will require to have `titleText` passed, but it's optional now.

Reply:
Agree with Raisa. This would be breaking. 

Also, shouldn't both be present since `aria-describedby` points to the target of the element with the id and `aria-description` is just describing what the element is with nothing pointing to it?

Reply:
> I think this is potentially breaking change, because now tooltip will require to have `titleText` passed, but it's optional now.

Does this mean `title` prop was optional? Why?

Reply:
> Also, shouldn't both be present since `aria-describedby` points to the target of the element with the id and `aria-description` is just describing what the element is with nothing pointing to it?

No, for similar reasons why we discourage using both `aria-labelledby` references and `aria-label` strings on the same element. Browsers follow an algorithm to define a 1) `accessible name` and 2) `accessible description` for an element. 

If `aria-describedby` is used on an element, then an `aria-description` string will be ignored and discarded by the browser.

Reply:
`titleText` is optional in the `useTooltip` hook, but not the `title` in the `Tooltip` component. `titleText` was optional in `useTooltip` because `type="describe"` did not use the config. This change makes it so `type="describe"` also uses the config. There is still the `type="muted"` that doesn't use the `titleText` config.

Now, `title` is required by `Tooltip` because it is the contents of the tooltip. `OverflowTooltip` does not use `title` because the contents of the target are the contents of the tooltip. An `OverflowTooltip` uses the `type="muted"` because a screen reader doesn't understand the concept of the ellipsis - to the screen reader, the full contents are exposed already. A screen dictation user will be able to target based on what content is already visible and not cut off - from there it is part of the design to make sure the most significant part of the target text is visible when overflowing.